### PR TITLE
Split RestProcessor into envelope and payload implementations

### PR DESCRIPTION
**Pull request by mapingo**

### CODE REVIEW

File: generators/rest-adapter-core/src/main/java/uk/gov/justice/services/adapter/rest/processor/EnvelopeResponseRestProcessor.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    private static final Logger LOGGER = getLogger(EnvelopeResponseRestProcessor.class);
+    private final JsonObjectEnvelopeConverter jsonObjectEnvelopeConverter;
+
+    EnvelopeResponseRestProcessor(final RestEnvelopeBuilderFactory envelopeBuilderFactory,
+                                  final JsonObjectEnvelopeConverter jsonObjectEnvelopeConverter) {
+        super(envelopeBuilderFactory, LOGGER);
+        this.jsonObjectEnvelopeConverter = jsonObjectEnvelopeConverter;
+    }
+
+    protected Response okResponseWith(final JsonEnvelope envelope) {
```
Comment (jaceko) [inline generators/rest-adapter-core/src/main/java/uk/gov/justice/services/adapter/rest/processor/EnvelopeResponseRestProcessor.java]: maybe a better name would be okResponseFrom or okResponseOf

File: generators/rest-adapter-core/src/main/java/uk/gov/justice/services/adapter/rest/processor/EnvelopeResponseRestProcessor.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        return status(OK)
+                .entity(jsonObjectEnvelopeConverter.fromEnvelope(envelope).toString())
```
Comment (jaceko) [inline generators/rest-adapter-core/src/main/java/uk/gov/justice/services/adapter/rest/processor/EnvelopeResponseRestProcessor.java]: shouldn't we use ObjectMapper here?

File: generators/rest-adapter-core/src/main/java/uk/gov/justice/services/adapter/rest/processor/PayloadResponseRestProcessor.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    private static final Logger LOGGER = getLogger(EnvelopeResponseRestProcessor.class);
+
+    PayloadResponseRestProcessor(final RestEnvelopeBuilderFactory envelopeBuilderFactory) {
+        super(envelopeBuilderFactory, LOGGER);
+    }
+
+    protected Response okResponseWith(final JsonEnvelope envelope) {
+        return status(OK)
+                .header(ID, envelope.metadata().id())
+                .entity(envelope.payloadAsJsonObject().toString())
```
Comment (jaceko) [inline generators/rest-adapter-core/src/main/java/uk/gov/justice/services/adapter/rest/processor/PayloadResponseRestProcessor.java]: shouldn't we use ObjectMapper here?

File: generators/rest-adapter-core/src/main/java/uk/gov/justice/services/adapter/rest/processor/EnvelopeResponseRestProcessor.java (hunk 1 above)
Comment (allanmckenzie) [inline generators/rest-adapter-core/src/main/java/uk/gov/justice/services/adapter/rest/processor/EnvelopeResponseRestProcessor.java]: +1

**Pull request merged**